### PR TITLE
fix(eslint): modify eslint config to catch missing stuff

### PR DESCRIPTION
**Pull request by CITguy**
Update ESLint config to catch stuff it previously missed.

### LGTM's
- [x] Dev LGTM
- [x] Dev LGTM

### CODE REVIEW

File: .eslintrc.json [hunk 1]
```diff
@@ -14,11 +14,14 @@
         "ShadyCSS": false
     },
     "rules": {
-        "brace-style": [2, "1tbs", { 
-            "allowSingleLine": true 
+        "array-bracket-newline": [2, { "minItems": 2 }],
```
Comment (CITguy) [inline .eslintrc.json]: Require newline after opening/before closing array bracket if there are 2 or more items (used in conjunction with `array-element-newline` below).

*Good*
```javascript
var a = [ 'one item' ];
var b = [
    'first item',
    'second item',
];
```

*Bad*
```javascript
var b = [ 'first item', 'second item' ];
```

File: .eslintrc.json [hunk 2]
```diff
@@ -14,11 +14,14 @@
...
+        "array-bracket-spacing": [2, "always"],
```
Comment (CITguy) [inline .eslintrc.json]: Ensure that there's always space after opening and before closing array brackets.

*Good*
```javascript
var a = [ 'foobar' ];
```

*Bad*
```javascript
var a = ['foobar'];
```

File: .eslintrc.json [hunk 3]
```diff
@@ -14,11 +14,14 @@
...
+        "array-element-newline": [2, "always"],
```
Comment (CITguy) [inline .eslintrc.json]: See comment on `array-bracket-newline`.

File: .eslintrc.json [hunk 4]
```diff
@@ -33,19 +36,20 @@
...
-            "allowTernary": true, 
+        "no-unused-expressions": [2, {
+            "allowTernary": true,
             "allowShortCircuit": true
         }],
         "no-unused-vars": 2,
         "no-use-before-define": [2, { "functions": false }],
         "object-curly-spacing": [2, "always"],
         "quotes": [2, "single"],
+        "semi": [2, "always"],
```
Comment (CITguy) [inline .eslintrc.json]: Always add semicolons. Do not rely on automatic semicolon insertion (ASI).

File: .eslintrc.json (hunk 1 above)
Comment (davismartin) [reply .eslintrc.json]: looks correct but shouldn't it have added new lines on
``` javascript
static get observedAttributes () {
       return ['open','position','trigger'];
}
```
in `hxTooltip.js`?

**Comment by davismartin** [comment]
+1 LGTM

**Pull request merged**